src/shared/cpu-set-util.c

```diff
@@ -34,6 +34,46 @@ char* cpu_set_to_string(const CPUSet *a) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char *cpu_set_to_range(const CPUSet *set) {
+        unsigned range_start = 0, range_end;
+        _cleanup_free_ char *str = NULL;
+        size_t allocated = 0, len = 0;
+        bool in_range = false;
+        int r;
+
+        for (unsigned i = 0; i < set->allocated * 8; i++) {
+                bool b = CPU_ISSET_S(i, set->allocated, set->set);
+
+                if (b) {
+                        if (in_range)
+                                range_end++;
+                        else {
+                                range_start = range_end = i;
+                                in_range = true;
+                        }
+                } else if (!b && in_range) {
```

Comment:
the `!b &&` part is redundant, given that this is the second branch and the first branch already covered the `b` case